### PR TITLE
Creation of the contrail-config role

### PR DESCRIPTION
**Pull request by Spredzy**

### CODE REVIEW

**Comment by Spredzy** [comment]
Packages and Services are from the paid version, will update the PR with the open source version name.

File: templates/config/basicauthusers.properties.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+test:test
+dhcp:dhcp
+visual:visual
+sensor:sensor
+# compliance testsuite users
+mapclient:mapclient
+helper:mapclient
+# This is a read-only MAPC
+reader:reader
+api-server:api-server
```
Comment (sbadia) [inline templates/config/basicauthusers.properties.erb]: for api-server, schema-transformer and svc-monitor you don't allow the password change ?

File: templates/config/log4j.properties.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Set root logger level to DEBUG and its only appender to CONSOLE
+log4j.rootLogger=TRACE, CONSOLE
+log4j.error
+log4j.logger.de.fhhannover.inform.irond.proc=TRACE, A1, A2
+log4j.additivity.de.fhhannover.inform.irond.proc=false
+log4j.appender.A1=org.apache.log4j.ConsoleAppender
+log4j.appender.A1.layout=org.apache.log4j.PatternLayout
+log4j.appender.A1.layout.ConversionPattern=%d [%t] %-5p %x - %m%n
+
```
Comment (sbadia) [inline templates/config/log4j.properties.erb]: Hum ? You are sure about `A2,A3,CONSOLE` ? This setup is very, very verbose :-) (and log all messages in console, supervisord and the ifmap log).

I use this setup for `ifmap-server`:

```
# Log configuration ifmap-server
#
log4j.rootLogger=INFO
log4j.error

log4j.logger.de.fhhannover.inform.irond.proc=INFO, A1
log4j.additivity.de.fhhannover.inform.irond.proc=false

log4j.appender.A1=org.apache.log4j.FileAppender
log4j.appender.A1.File=/var/log/contrail/ifmap-server.log
log4j.appender.A1.layout=org.apache.log4j.PatternLayout
log4j.appender.A1.layout.ConversionPattern=%d [%t] %-5p %x - %m%n
```

**Comment by sbadia** [comment]
Just some comments, but otherwise it's ok

File: templates/config/basicauthusers.properties.erb (hunk 1 above)
Comment (Spredzy) [inline templates/config/basicauthusers.properties.erb]: @sbadia this feature will be added on a subsequent release.

**Pull request merged**